src/libtls/tls_peer.c

```diff
@@ -1426,37 +1504,45 @@ static status_t send_certificate(private_tls_peer_t *this,
 	}
 	enumerator->destroy(enumerator);
 
+	/* certificate request context as described in RFC 8446, section 4.4.2 */
+	if (this->tls->get_version_max(this->tls) > TLS_1_2)
+	{
+		writer->write_uint8(writer, 0);
+	}
+
 	/* generate certificate payload */
 	certs = bio_writer_create(256);
-	if (this->peer)
+	cert = this->peer_auth->get(this->peer_auth, AUTH_RULE_SUBJECT_CERT);
+	if (cert)
 	{
-		cert = this->peer_auth->get(this->peer_auth, AUTH_RULE_SUBJECT_CERT);
-		if (cert)
+		if (cert->get_encoding(cert, CERT_ASN1_DER, &data))
+		{
+			DBG1(DBG_TLS, "sending TLS peer certificate '%Y'",
```

Comment:
Maybe change this to
```suggestion
			DBG1(DBG_TLS, "sending TLS client certificate '%Y'",
```